asv_bench/benchmarks/index_object.py

```diff
@@ -1,207 +1,178 @@
-from .pandas_vb_common import *
+import numpy as np
+import pandas.util.testing as tm
+from pandas import date_range, DatetimeIndex, Index, MultiIndex, RangeIndex
+
+from .pandas_vb_common import setup  # noqa
 
 
 class SetOperations(object):
+
     goal_time = 0.2
 
     def setup(self):
-        self.rng = date_range('1/1/2000', periods=10000, freq='T')
-        self.rng2 = self.rng[:(-1)]
+        self.dates_left = date_range('1/1/2000', periods=10000, freq='T')
+        self.dates_right = self.dates_left[:(-1)]
 
-        # object index with datetime values
-        if (self.rng.dtype == object):
-            self.idx_rng = self.rng.view(Index)
-        else:
-            self.idx_rng = self.rng.astype(object)
-        self.idx_rng2 = self.idx_rng[:(-1)]
+        fmt = '%Y-%m-%d %H:%M:%S'
+        self.date_str_left = Index(self.dates_left.strftime(fmt))
+        self.date_str_right = self.date_str_left[:-1]
```

Comment:
you could parametrize these over the actual operations (e.g. difference, intersection, etc).  if you think its worth it.

Reply:
There is one black sheep benchmark, `time_datetime_difference_disjoint`, that makes it a little more work to parametrize, but if i separate it out to a new class the rest should be easy to parametrize